test/src/graph/nodes/CompactPointsNodeTest.cpp

```diff
@@ -1,24 +1,135 @@
+    ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(usePointsNode, compactNode));
+    ASSERT_RGL_SUCCESS(rgl_graph_run(usePointsNode));
+
+    int32_t hitpointCount, pointSize;
+    ASSERT_RGL_SUCCESS(rgl_graph_get_result_size(compactNode, RGL_FIELD_XYZ_F32, &hitpointCount, &pointSize));
+    EXPECT_EQ(hitpointCount, 0);
+}
+
+TEST_P(CompactPointsNodeTest, should_not_remove_any_point_when_all_hit)
+{
+    int pointsCount = GetParam();
+
+    createTestUsePointsNode(pointsCount, identityTestTransform, HitPointDensity::ALL_HIT);
+    ASSERT_RGL_SUCCESS(rgl_node_points_compact(&compactNode));
+    ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(usePointsNode, compactNode));
+    ASSERT_RGL_SUCCESS(rgl_graph_run(usePointsNode));
+
+    int32_t hitpointCount, pointSize;
+    ASSERT_RGL_SUCCESS(rgl_graph_get_result_size(compactNode, RGL_FIELD_XYZ_F32, &hitpointCount, &pointSize));
+    EXPECT_EQ(hitpointCount, pointsCount);
+}
+
+TEST_P(CompactPointsNodeTest, should_remove_as_many_points_as_non_hit)
+{
+    int pointsCount = GetParam();
+
+    createTestUsePointsNode(pointsCount, identityTestTransform, HitPointDensity::RANDOM);
+    ASSERT_RGL_SUCCESS(rgl_node_points_compact(&compactNode));
+    ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(usePointsNode, compactNode));
+    ASSERT_RGL_SUCCESS(rgl_graph_run(usePointsNode));
+
+    int32_t hitpointCount, pointSize;
+    ASSERT_RGL_SUCCESS(rgl_graph_get_result_size(compactNode, RGL_FIELD_XYZ_F32, &hitpointCount, &pointSize));
+    EXPECT_EQ(hitpointCount, pointsCount - randomNonHitCount);
+}
+
+TEST_P(CompactPointsNodeTest, should_not_change_hit_points)
+{
+    int pointsCount = GetParam();
+
+    createTestUsePointsNode(pointsCount, identityTestTransform, HitPointDensity::RANDOM);
+    std::vector<TestPointStruct> expectedHitPoints = separateHitPoints();
+    if(expectedHitPoints.empty())
+        return;
+
+    ASSERT_RGL_SUCCESS(rgl_node_points_compact(&compactNode));
+    ASSERT_RGL_SUCCESS(rgl_graph_node_add_child(usePointsNode, compactNode));
+    ASSERT_RGL_SUCCESS(rgl_graph_run(usePointsNode));
+
+    for (auto field : pointFields) {
+        int32_t outCount, outSizeOf;
+        EXPECT_RGL_SUCCESS(rgl_graph_get_result_size(compactNode, field, &outCount, &outSizeOf));
+        EXPECT_EQ(outCount, expectedHitPoints.size());
+        EXPECT_EQ(outSizeOf, getFieldSize(field));
+
+        switch(field) {
+            case XYZ_F32: {
+                std::vector<::Field<XYZ_F32>::type> outData;
+                outData.resize(outCount);
+                EXPECT_RGL_SUCCESS(rgl_graph_get_result_data(compactNode, field, outData.data()));
+                for (int i = 0; i < outCount; ++i) {
+                    EXPECT_NEAR(expectedHitPoints.at(i).xyz[0], outData.at(i)[0], EPSILON_F);
+                    EXPECT_NEAR(expectedHitPoints.at(i).xyz[1], outData.at(i)[1], EPSILON_F);
+                    EXPECT_NEAR(expectedHitPoints.at(i).xyz[2], outData.at(i)[2], EPSILON_F);
```

Comment:
Consider using `EXPECT_EQ`. I'd expect the data to remain unchanged bit-wise.

Reply:
If it does not pass, consider comparing the data from the parent node. 
You can add a field filled with std::iota to help find indices to parent data after compaction.